### ISSUE TITLE
rule: `unsafe-copy`

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#201]
## Description
Passing types that store a pointer and capacity can lead to memory leaks. 

## Examples
Examples of **incorrect** code for this rule
```zig
fn foo(this: *Foo) std.ArrayList(u32) {
  return this.list;
}
```
If the caller appends any items to the returned list, it is most likely a mistake that will cause a memory leak. 

Examples of **correct** code for this rule
```zig

fn foo(this: *Foo) *std.ArrayList(u32) {
  return &this.list;
}

```

A couple of extra notes:
* should check list, heap, and ArenaAllocator types
* should check parameters and returned values
* returning a new instance by value should not be a violation
